ocaml/lambda/lambda.mli

```diff
@@ -332,6 +332,39 @@ type primitive =
      if the value is locally allocated *)
   (* Fetching domain-local state *)
   | Pdls_get
+    (* In-place reuse of heap blocks.
+     Each of these takes the allocation to be reused as first argument.
+     The following arguments correspond to the fields that are Reuse_set. *)
+  | Preuseblock of
```

Comment:
As @anfelor and I discussed over Slack, I don't currently see the need to have 4 separate primitives here. The overwriting action for all the different kinds of blocks should be the same, I think. This is really a question for @mshinwell or the folks at OCamlPro to answer than @anfelor, but I'm wondering if all of this could be collapsed to one `Preuseblock`. Do we need `mutable_flag`? Do we need `alloc_mode`? Without more knowledge of the middle/back ends, I can't be sure, but it's not obvious to me why we'd need these here.

Reply:
As just discussed in our meeting, we want to keep all this information in since it will make the implementation in the backend easier.